File: app/src/main/java/me/imzack/app/end/injector/component/TypeCreationComponent.java
package me.imzack.app.end.injector.component;

import me.imzack.app.end.injector.module.TypeCreationPresenterModule;
import me.imzack.app.end.injector.scope.ActivityScope;
import me.imzack.app.end.view.activity.TypeCreationActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = TypeCreationPresenterModule.class)
public interface TypeCreationComponent {

    void inject(TypeCreationActivity typeCreationActivity);
}

File: app/src/main/kotlin/me/imzack/app/end/injector/component/TypeCreationComponent.kt
package me.imzack.app.end.injector.component

import dagger.Component
import me.imzack.app.end.injector.module.TypeCreationPresenterModule
import me.imzack.app.end.injector.scope.ActivityScope
import me.imzack.app.end.view.activity.TypeCreationActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(TypeCreationPresenterModule::class))
interface TypeCreationComponent {

    fun inject(typeCreationActivity: TypeCreationActivity)
}
